### PR TITLE
Feature/genesys mocks

### PR DESCRIPTION
**Pull request by ipekbayrak**

### CODE REVIEW

File: protractor.conf.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const protractorConfig = {
 
   // list of files / patterns to load in the browser
   specs: [
-    'tests/base.spec.js'
+    'tests/base.genesys.spec.js'
```
Comment (shadji) [inline protractor.conf.js:22]: We probably need all tests to be enabled in main

File: tests/lib/mockProxy.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+          'Access-Control-Allow-Methods': 'GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH',
+          Allow: 'GET, PUT, POST, DELETE, HEAD, OPTIONS, PATCH',
+          'access-control-max-age': 2592000
+        };
+        http.createServer(function (req, res) {
+          log.debug("REQUEST rcvd===> Method:" + req.method + " PATH:" + req.url);
+
+          const mresp = MockProxy.mockObjects.find(function (element) {
+            const rule = match(element.rule.path);
+            return rule(req.url) && rule.path === req.path && element.rule.method === req.method;
```
Comment (shadji) [inline tests/lib/mockProxy.js:147]: Why? && rule.path === req.path

File: tests/lib/cert/myCA.pem [hunk 3]
```diff
@@ -0,0 +1,25 @@
+-----BEGIN CERTIFICATE-----
```
Comment (shadji) [inline tests/lib/cert/myCA.pem:1]: Plese remove all not needed stuff, and change names to 'fakeselfcreated' crt/key

File: tests/lib/cert/mypurecloud.com.au.crt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+-----BEGIN CERTIFICATE-----
```
Comment (shadji) [inline tests/lib/cert/mypurecloud.com.au.crt:1]: remove !

File: tests/lib/mockProxy.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+   * @returns promise
+   */
+  startSSlProxyServer (port = 9001) {
+    return new Promise(function (resolve, reject) {
+      try {
+        const sslProxy = httpProxy.createProxyServer({
+          target: 'http://localhost:' + port,
+          ssl: {
+            key: fs.readFileSync('./tests/lib/cert/mypurecloud.com.au.key', 'utf8'),
+            cert: fs.readFileSync('./tests/lib/cert/mypurecloud.com.au.crt', 'utf8')
```
Comment (shadji) [inline tests/lib/mockProxy.js:45]: never use names of the third party companies plz. make sure names explicitly describe the purpose of this selfsigned fake certificates.

File: tests/lib/mockProxy.js (hunk 2 above)
Comment (ipekbayrak) [reply tests/lib/mockProxy.js]: removed

File: tests/lib/mockProxy.js (hunk 5 above)
Comment (ipekbayrak) [reply tests/lib/mockProxy.js]: renamed

**Pull request merged**